iceoryx_posh/source/capro/service_description.cpp

```diff
@@ -234,15 +227,10 @@ Interfaces ServiceDescription::getSourceInterface() const noexcept
     return m_interfaceSource;
 }
 
-/// @todo remove AnyInstanceString's no invalid ServiceDescription is allowed
 bool ServiceDescription::isValid() const noexcept
 {
-    return !(m_serviceString == iox::capro::InvalidString
-             || m_serviceString == capro::IdString_t(cxx::TruncateToCapacity, iox::capro::AnyServiceString)
-             || m_instanceString == iox::capro::InvalidString
-             || m_instanceString == capro::IdString_t(cxx::TruncateToCapacity, iox::capro::AnyInstanceString)
-             || m_eventString == iox::capro::InvalidString
-             || m_eventString == capro::IdString_t(cxx::TruncateToCapacity, iox::capro::AnyEventString));
+    return !(m_serviceString == iox::capro::InvalidIdString || m_instanceString == iox::capro::InvalidIdString
+             || m_eventString == iox::capro::InvalidIdString);
```

Comment:
This is just a personal preference and you can keep this as is if you want but I find it always hard to grasp when the result of a longer expression is negated. In this case it would be easier for me to understand if it would be `return (m_serviceString != iox::capro::InvalidIdString && ...)`

Reply:
Alrighty, for me it's the other way round :upside_down_face: